### PR TITLE
feat: re-enable jira

### PR DESCRIPTION
**Pull request by JoeHCQ1**
Jira is re-enabled and running in the dev cluster.

Also, core is updated and uds cli are updated. 

Docs are updated to reflect updated versions of core and jira.

### CODE REVIEW

**Comment by JoeHCQ1** [comment]
It works:

**Comment by JoeHCQ1** [comment]
That is, it works with the changes I made to the Jira package: https://github.com/defenseunicorns/uds-package-jira/pull/49

**Comment by JoeHCQ1** [comment]
Not that we need the update per-se, it's specific to my particular rendition of the dev/test uds-config.yaml. But if I don't update the uds config yet, then it works with the existing version of the jira package.

**Comment by JoeHCQ1** [comment]
Note that after adding in the netpol to allow egress to the DB, the next blocker was that the dev db for Jira still had an old password, not the password in the uds-config. To reset the credentials, you need to remove cached values from Jira. In a dev scenario, removing Jira, and it's PVCs seemed to be the best option.

Also, Jira can take a long time to fully come up.

**Comment by JoeHCQ1** [comment]
testing this again locally with reduced changes per simplification given clarity around what changes were needed.

**Comment by JoeHCQ1** [comment]
And it's broken again... so frustrating.

**Comment by JoeHCQ1** [comment]
Let's get the improvements to-date merged, I'm going to work on something else.

File: bundles/uds-core-swf/uds-bundle.yaml [hunk 1]
```diff
@@ -135,20 +135,6 @@ packages:
                   cpu: 1
                   memory: 2Gi
           values:
-            - path: initContainers
-              value:
-                - name: velero-plugin-for-aws
```
Comment (jacobbmay) [inline bundles/uds-core-swf/uds-bundle.yaml:140]: Is this one not still needed? I have a local branch deploying updated uds-core and I only had to remove the csi plugin, but it is still functioning with the aws plugin being deployed.

Comment (JoeHCQ1) [reply bundles/uds-core-swf/uds-bundle.yaml:140]: I was told in slack to remove both iirc. Also, core now includes these or at least one of them so this choice on our end is either a null-act or a deviation from what core expects in terms of versions.

Comment (anthonywendt) [reply bundles/uds-core-swf/uds-bundle.yaml:140]: Yeah It can be removed with the recent versions as far as I'm aware.

Comment (JoeHCQ1) [reply bundles/uds-core-swf/uds-bundle.yaml:140]: Here's the PR I was sent to: https://github.com/defenseunicorns/uds-core/pull/486

Comment (JoeHCQ1) [reply bundles/uds-core-swf/uds-bundle.yaml:140]: Here's the slack message: https://defense-unicorns.slack.com/archives/C06QJAUHWFN/p1726853160622109

Comment (JoeHCQ1) [reply bundles/uds-core-swf/uds-bundle.yaml:140]: I was told specifically to delete these lines: https://github.com/defenseunicorns/uds-bundle-software-factory-nutanix/blob/main/bundles/uds-core-swf/uds-bundle.yaml#L146-L151

File: tasks/deploy.yaml [hunk 2]
```diff
@@ -20,20 +20,20 @@ tasks:
...
+      build_dir:
         default: ./build
         description: Input for the path to the build directory containing the uds bundle to deploy
-      config-dir:
+      config_dir:
         default: ./scratch/configs/dev
         description: Input for the path to the directory containing the uds-config.yaml to use for deploying the bundle
     actions:
-      - cmd: docker run --rm -e ARCH=${ARCH} -e BUNDLE_VERSION=${BUNDLE_VERSION} -e UDS_CONFIG=/configs/uds-config.yaml -e KUBECONFIG=/configs/kubeconfig -v /tmp:/tmp -v ${INPUT_CONFIG_DIR}:/configs -v ${INPUT_BUILD_DIR}:/build ${DEPLOY_IMAGE} bash -c 'uds deploy /build/uds-bundle-software-factory-nutanix-${ARCH}-${BUNDLE_VERSION}.tar.zst --confirm'
+      - cmd: UDS_CONFIG=${{ .inputs.config_dir }}/uds-config.yaml KUBECONFIG=${{ .inputs.config_dir }}/kubeconfig uds deploy ${{ .inputs.build_dir }}/uds-bundle-software-factory-nutanix-${ARCH}-${BUNDLE_VERSION}.tar.zst --confirm
```
Comment (jacobbmay) [inline tasks/deploy.yaml:30]: This should still be running in the build-harness container to ensure deploys are run in similar environments for everyone and to eliminate some of the "works on my machine" scenarios.

Comment (JoeHCQ1) [reply tasks/deploy.yaml:30]: This is a perspective I disagree with for a few reasons:

1. Using docker like this increases the visual complexity of the command dramatically. This slows down anyone who is either less familiar with docker or just not familiar with the goal - re-ordering the flags would help but it's still hard to parse.
2. This creates what I'd call a hyper-sanitized environment that enables fragility - this shouldn't be necessary and customers will probably not be quite so uniform - I'd welcome this low level of chaos to improve engineering.
3. UDS is already our common foundation - as long as we're running similar versions, it's a problem (that I'd want discovered and fixed) if minor version changes break stuff. This use of docker replaces UDS with a sort of "magic VM". Back to the benefits of low-level chaos.
4. By way of analogy, it has all the strengths and weaknesses of a magic VM.

Basically, if we want to eliminate "it works on my machine" we should use multiple machines to spot points of fragility. Agreeing to all use one machine is enabling, not correcting the problem.

I'd be more in favor of this approach if this were a C++ app - those are miserable to encapsulate. However, Go is great at dependency management and cross-OS performance, so this layer of encapsulation seems unnecessary.

Have we had any problems from minor differences in UDS versions? If we are having problems from that, wouldn't we need to then run _all_ UDS commands via the build image or be deploying with one image and administrating with another?

Comment (JoeHCQ1) [reply tasks/deploy.yaml:30]: One example of what was hard to parse: the expected file structure. With thing getting mounted into docker and then referenced by internal paths it was not obvious where I needed to put stuff locally to get it to work. Running the command straight out makes that really clear - it's all in `scratch/env/<kubeconfig/uds-config.yaml>`

Comment (jacobbmay) [reply tasks/deploy.yaml:30]: The reason this was added is because even if we are both using the same version of uds cli there can be (and have been) different behaviors caused by environmental differences.

A simple example (that has actually happened. It isn't just a hypothetical): You are running deploys from your macbook pro cli while developing. Someone else gets your changes but they are running their deploys in a pipeline using some random linux image. Your changes added some base64 command in a zarf action that uses a specific flag that you assume will be available because it was available on your mac. The person running the deploy in a pipeline is using an image that has the base64 command, but it doesn't have the flag you are depending on. The deploy breaks for them even though it worked for you.

By using the same image to run deploys in all locations you avoid these kinds of issues.

Comment (jacobbmay) [reply tasks/deploy.yaml:30]: Because zarf packages can have actions that run arbitrary cli commands it makes sense to run deploys in a container to ensure that the commands needed are always there.

Comment (JoeHCQ1) [reply tasks/deploy.yaml:30]: Got it - Zarf is our C++ ;). That also explains why the deploy command specifically might be encapsulated w/o the other ones being encapsulated.

**Review by jacobbmay — APPROVED** [review]
LGTM
